File: internal/game/states/dead.go
package states

import (
	"github.com/slycrel/slycrel/internal/game"
	"github.com/slycrel/slycrel/internal/mechanics"
)

// DeadState handles player death - offers to start a new day or quit.
type DeadState struct{}

func (DeadState) ID() game.StateID { return "dead" }

func (DeadState) Enter(s *game.Session) {
	s.IO.Cr()
	choice := s.IO.LettersPrompt("[N]ew Day (resurrect & play again) or [Q]uit?", "NQ", 1, true, true)

	if choice == "N" {
		// Resurrect and start fresh day
		mechanics.NewDayForUser(s.Character)
		s.Character.LastOn = todayDate()
		s.Store.SaveCharacter(s.Character)
		s.IO.Cr()
		s.IO.Outln("A new day dawns... you have been restored.", true, 3)
		s.IO.Cr()
		s.SetNextState("begin")
	} else {
		s.IO.Outln("Now Returning to the BBS", true, 5)
		s.IO.Cr()
		s.Quit()
	}
}

File: clients/web/game/states/dead.js
import { newDayForUser } from '../../mechanics/resurrection.js';
import { saveCharacter } from '../../store/local.js';
import { BeginState } from './begin.js';

// DeadState — port of internal/game/states/dead.go.
// Offers a "new day" resurrection (the only way back in a browser-only
// build — we don't model elapsed real-time the way the Go server does).
export class DeadState {
  async enter(session) {
    const { io } = session;
    io.cr();
    const choice = await io.lettersPrompt('[N]ew Day (resurrect & play again) or [Q]uit?', 'NQ');
    io.cr();
    if (choice === 'N') {
      newDayForUser(session.character);
      saveCharacter(session.character);
      io.println('A new day dawns... you have been restored.', 3);
      io.cr();
      session.setNext(new BeginState());
    } else {
      io.println('Now Returning to the BBS', 5);
      io.cr();
      session.quit();
    }
  }
}
